source/controllers/users.ts

```diff
@@ -1,85 +1,61 @@
-import { Request, Response, NextFunction } from 'express';
-import Logging from '../configs/logging';
-import Bcryptjs, { hash } from 'bcryptjs';
-import Mongoose from 'mongoose';
-import Users from '../models/users';
-import SignJwt from '../functions/sign-JWT'
-
-
-
 
-const NAMESPACE = 'User';
-const validateToken = (req: Request, res: Response, next: NextFunction) => {
-    Logging.info(NAMESPACE, 'Token validated');
 
-    return res.status(200).json({
-        message: 'Authorized'
-    });
-};
+import Bcryptjs from 'bcryptjs';
+import Mongoose from 'mongoose';
+import Users from '../models/users';
+import SignJwt from '../functions/sign-jwt';
 
-const addUser = (req: Request, res: Response, next: NextFunction) => {
-    let { username, password } = req.body;
 
-    Bcryptjs.hash(password, 10, (hashError, hash) => {
-        if (hashError) {
-            return res.status(500).json({
-                message: hashError.message,
-                error: hashError
-            });
-        }
+const addUser = async(ctx ,next) => {
+    let { username, password } = ctx.request.body;
 
-        //insert user to DB
-        const user = new Users({
-            _id: new Mongoose.Types.ObjectId(),
-            username,
-            password:hash
-        })
+    const user = new Users({
+        _id: new Mongoose.Types.ObjectId(),
+        username,
+        password: await Bcryptjs.hash(password, 10)
+    })
 
-        return user.save()
-        .then(user => {
-            return res.status(201).json({
-                user
-            })
-        })
-        .catch(error => {
-            return res.status(500).json({
-                message:error.message,
-                error
-            }) 
-        })
-    });
+    const find = await Users.find({username})  
+
+    if(find.length!=0){
+        ctx.status = 400
+        ctx.body='User already exist'
+    }else{
+        const res = await user.save()
+    
+        ctx.status=200
+        ctx.body={res}
+    }
+ 
 };
 
-const login = (req: Request, res: Response, next: NextFunction) => {
-    let {username, password} = req.body;
+const login = async(ctx ,next) => {
+    let {username, password} = ctx.body;
 
-    Users.find({username})
-    .exec()
+    await Users.find({username})
     .then((users) => {
         if(users.length !==1 ){
-            return res.status(401).json({
-                message:'Unauthorized'
-            })
+            ctx.status = 401
+            ctx.message='Unauthorized'
+
         }
```

Comment:
Please use `async/await`, rather than `thenables`

Like you can assign `await User.findOne({ username })` into a variable and wrap it in a `try..catch`:
```ts
try {
  const user = await UserModel.findOne({ username })
  // ...
} catch (error) {
 // handle error
}
```

Reply:
copy sir. di ko muna ito isa sali sa push may ibahin pa ko